### PR TITLE
Bug 1726726 - Implement a Store for the web platform

### PR DESCRIPTION
**Pull request by brizental**
### Pull Request checklist ###

- [x] **Quality**: Make sure this PR builds and runs cleanly.
  - Inside the `glean/` folder, run:
    - `npm run test` Runs _all_ tests
    - `npm run lint` Runs _all_ linters
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes a changelog entry to `CHANGELOG.md` or an explanation of why it does not need one
- [x] **Documentation**: This PR includes documentation changes, an explanation of why it does not need that or a follow-up bug has been filed to do that work
  - https://bugzilla.mozilla.org/show_bug.cgi?id=1736352

### CODE REVIEW

File: glean/src/platform/browser/web/storage.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+          // but let's leave it here to make TypeScript happy and to be extra careful.
+          throw new Error("IMPOSSIBLE: Unable to perform database transaction. Database is not initialized.");
+        }
+
+        const completed = new Promise<void>((resolve, reject) => {
+          transaction.oncomplete = () => resolve();
+          transaction.onerror = transaction.onabort = e => reject(e);
+        });
+
+        const store = transaction?.objectStore(STORE_NAME);
```
Comment (Dexterp37) [inline glean/src/platform/browser/web/storage.ts:85]: I would have expected the `?` to not be required here, given the check on line 74!

File: glean/src/platform/browser/web/storage.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    }
+
+    return IDBWrapper._instance;
+  }
+
+  private static async withDatabase(fn: () => Promise<void>) {
+    if (!IDBWrapper.instance.db) {
+      return new Promise<void>((resolve, reject) => {
+        // `self` will work in Node.js and in the browser.
+        // We want this to work in Node.js **only** for testing purposes.
```
Comment (Dexterp37) [inline glean/src/platform/browser/web/storage.ts:38]: What do you mean here? What's `self`?

**Review by Dexterp37 — CHANGES_REQUESTED** [review]

File: glean/src/platform/browser/web/storage.ts (hunk 1 above)
Comment (brizental) [reply glean/src/platform/browser/web/storage.ts:85]: Tell that to the typeScript checker!

Comment (brizental) [reply glean/src/platform/browser/web/storage.ts:85]: (Jokes aside, maybe this is leftover from developement. I'll check if TS really complains).

**Review by Dexterp37 — APPROVED** [review]

**Pull request merged**